src/agent/src/rpc.rs

```diff
@@ -1538,7 +1594,12 @@ fn do_copy_file(req: &CopyFileRequest) -> Result<()> {
         PathBuf::from("/")
     };
 
-    fs::create_dir_all(dir.to_str().unwrap()).or_else(|e| {
+    fs::create_dir_all(
+        dir.to_str()
+            .ok_or("failed to get directory to create")
+            .map_err(|e| anyhow!(e))?,
+    )
+    .or_else(|e| {
```

Comment:
So, this code is taking a `Path`, converting to a string (which might fail) then passing it to `std::fs` function which actually wants a `Path`.  You should be able to just remove the `to_str().unwrap()` instead.